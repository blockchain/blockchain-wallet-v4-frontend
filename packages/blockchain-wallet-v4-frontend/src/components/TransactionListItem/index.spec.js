import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import ListItemContainer from './index'
import configureStore from 'redux-mock-store'

jest.mock('./template', () => () => { return (<div/>) })

const store = configureStore([])({})
const tx = { hash: '123abc' }

describe('ListItemContainer', () => {
  it('renders correctly', () => {
    const component = mount(<ListItemContainer minConfirmations={3} store={store} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
  describe('handleCoinToggle()', () => {
    it('calls preferencesActions.toggleCoinDisplayed', () => {
      const component = shallow(<ListItemContainer minConfirmations={3} store={store} />)
      const instance = component.dive().instance()
      const spy = jest.spyOn(instance.props.preferencesActions, 'toggleCoinDisplayed')
      instance.handleCoinToggle()
      expect(spy).toHaveBeenCalled()
    })
  })
  describe('handleEditDescription()', () => {
    it('handles eth tx notes', () => {
      const component = shallow(<ListItemContainer minConfirmations={3} transaction={tx} coin='ETH' store={store} />)
      const instance = component.dive().instance()
      const spy = jest.spyOn(instance.props.ethereumActions, 'setTxNotesEthereum')
      instance.handleEditDescription()
      expect(spy).toHaveBeenCalled()
    })
    it('handles btc tx notes', () => {
      const component = shallow(<ListItemContainer minConfirmations={3} transaction={tx} coin='BTC' store={store} />)
      const instance = component.dive().instance()
      const spy = jest.spyOn(instance.props.walletActions, 'setTransactionNote')
      instance.handleEditDescription()
      expect(spy).toHaveBeenCalled()
    })
  })
})
