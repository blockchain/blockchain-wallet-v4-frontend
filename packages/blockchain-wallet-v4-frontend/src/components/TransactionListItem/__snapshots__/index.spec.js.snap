// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemContainer renders correctly 1`] = `
<Connect(ListItemContainer)
  minConfirmations={3}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ListItemContainer
    bchActions={
      Object {
        "createMetadataBch": [Function],
        "fetchMetadataBch": [Function],
        "fetchMetadataBchFailure": [Function],
        "fetchMetadataBchLoading": [Function],
        "fetchMetadataBchSuccess": [Function],
        "setAccountArchived": [Function],
        "setAccountLabel": [Function],
        "setDefaultAccountIdx": [Function],
        "setTxNotesBch": [Function],
      }
    }
    ethereumActions={
      Object {
        "createMetadataEthereum": [Function],
        "fetchMetadataEthereum": [Function],
        "fetchMetadataEthereumFailure": [Function],
        "fetchMetadataEthereumLoading": [Function],
        "fetchMetadataEthereumSuccess": [Function],
        "setLatestTxEthereum": [Function],
        "setLatestTxTimestampEthereum": [Function],
        "setTxNotesEthereum": [Function],
        "updateMetadataEthereum": [Function],
      }
    }
    minConfirmations={3}
    preferencesActions={
      Object {
        "hideKycCompleted": [Function],
        "setBitcoinCashWelcome": [Function],
        "setBitcoinWelcome": [Function],
        "setCulture": [Function],
        "setEtherWelcome": [Function],
        "setLanguage": [Function],
        "setTheme": [Function],
        "toggleCoinDisplayed": [Function],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    walletActions={
      Object {
        "createLegacyAddress": [Function],
        "deleteHdAddressLabel": [Function],
        "deleteLegacyAddress": [Function],
        "deleteWrapper": [Function],
        "refreshWrapper": [Function],
        "setAccountArchived": [Function],
        "setAccountLabel": [Function],
        "setAddressArchived": [Function],
        "setAutoLogout": [Function],
        "setDefaultAccountIdx": [Function],
        "setHdAddressLabel": [Function],
        "setLegacyAddressLabel": [Function],
        "setMainPassword": [Function],
        "setPayloadChecksum": [Function],
        "setSyncPubKeys": [Function],
        "setTransactionNote": [Function],
        "setWrapper": [Function],
        "verifyMnemonic": [Function],
      }
    }
  >
    <Component
      handleCoinToggle={[Function]}
      handleEditDescription={[Function]}
      minConfirmations={3}
    >
      <div />
    </Component>
  </ListItemContainer>
</Connect(ListItemContainer)>
`;
